aptible/resource_app.go

```diff
@@ -641,72 +641,91 @@ func scaleServices(d *schema.ResourceData, meta interface{}) error {
-		hasOtherChanges := false
-		for key, newValue := range serviceData {
+		shouldScale := false
+		for key, newValue := range serviceInterface {
 			if key == "force_zero_downtime" || key == "simple_health_check" || key == "autoscaling_policy" || key == "service_sizing_policy" {
 				continue // Skip checking these keys. Nothing to do with manual scaling
 			}
 
 			if oldServiceData == nil || oldServiceData[key] != newValue {
-				hasOtherChanges = true
+				shouldScale = true
 				break
 			}
 		}
-
-		if hasOtherChanges {
-			filteredServices = append(filteredServices, serviceData)
+		if !shouldScale {
+			return nil
 		}
-	}
 
-	for _, s := range filteredServices {
-		// https://stackoverflow.com/a/74383278
-		service := s
-		serviceInterface := service.(map[string]interface{})
 		g.Go(func() error {
-			memoryLimit := int64(serviceInterface["container_memory_limit"].(int))
-			containerProfile := serviceInterface["container_profile"].(string)
-			containerCount := int64(serviceInterface["container_count"].(int))
+			memoryLimit := serviceInterface["container_memory_limit"]
+			containerProfile := serviceInterface["container_profile"]
+			containerCount := serviceInterface["container_count"]
 			processType := serviceInterface["process_type"].(string)
 
 			log.Printf(
 				"Updating %s service to count: %d, limit: %d, and container profile: %s\n",
 				processType, containerCount, memoryLimit, containerProfile,
 			)
-			service, err := client.GetServiceForAppByName(appID, processType)
-			if err != nil {
-				log.Println("There was an error when finding the service \n[ERROR] -", err)
-				return generateErrorFromClientError(err)
+			service := findApiServiceByName(apiServices, processType)
+			if service == nil {
+				return fmt.Errorf("there was an error when finding the service: %s", processType)
+			}
+
+			payload := aptibleapi.NewCreateOperationRequest("scale")
+			if oldServiceData["container_count"] != containerCount && containerCount != -1 {
+				log.Println("Updating containers-count")
+				payload.SetContainerCount(int32(containerCount.(int)))
 			}
-			err = client.ScaleService(service.ID, containerCount, memoryLimit, containerProfile)
+			log.Printf("memory limit old: %s, new: %s", oldServiceData["container_memory_limit"], memoryLimit)
+			if oldServiceData["container_memory_limit"] != memoryLimit && memoryLimit != -1 {
+				log.Println("Setting memory limit")
+				payload.SetContainerSize(int32(memoryLimit.(int)))
+			}
+			if oldServiceData["container_profile"] != containerProfile {
+				payload.SetInstanceProfile(containerProfile.(string))
+			}
+			resp, _, err := client.OperationsAPI.CreateOperationForService(ctx, service.Id).CreateOperationRequest(*payload).Execute()
 			if err != nil {
 				log.Println("There was an error when scaling the service \n[ERROR] -", err)
-				return generateErrorFromClientError(err)
+				return err
 			}
+
+			_, err = legacy.WaitForOperation(int64(resp.Id))
+			if err != nil {
```

Comment:
You could just return `err` here and it would do the same thing as lines 708-710